### ISSUE TITLE
./node_modules/.bin/npx not found :(

### ISSUE THREAD

**Issue by zkat** [zacanger/dist-upgrade#1]
```
➜ npx dist-upgrade --help
npx: installed 508 in 8.375s
/Users/zkat/.npm/_npx/67934/bin/dist-upgrade: line 3: ./node_modules/.bin/npx: No such file or directory
```

Unfortunately, there's no easy, surefire way to get the path to a dep's binary from a shell script. :\

**Comment by zautumnz** [comment]
Thanks for pointing that out! I _believe_ something like this would handle a most cases, but it's totally possible I'm missing something important. Do you have any thoughts on that?

```shell
#!/bin/sh

WHICH=`which npx`
NPMBIN=`npm bin`/npx
NPX=${WHICH:-$NPMBIN}

$NPX n latest && npm i -g npm@next
```

**Comment by zkat** [comment]
This still won't run the npx included with `dist-upgrade`.

The issue here is that your shell script can't figure out where _it_ is -- it runs from `pwd`. Since the bash script is symlinked, it won't know where your original node_modules is.

Something like this will do the trick though:
```
#!/usr/bin/env node

const cp = require('child_process')
const path = require('path')

const NPX_BIN = path.join(__dirname, 'node_modules', '.bin', 'npx')
cp.spawn(`${NPX_BIN} n latest && npm i -g npm@next`, [], {
  stdio: 'inherit',
  shell: true
})
```

'cause `__dirname` works right.

**Comment by zautumnz** [comment]
Thanks so much! I feel like a bit of a ditz for not realising that. Fixed in 0.2.0.
